internal/evm/gateway.go

```diff
@@ -0,0 +1,99 @@
+	gatewayCommand.AddToParent(Cmd)
+}
+
+type gatewayFlag struct {
+	DatabaseDir        string `flag:"database-dir" default:"./db" info:"path to the directory for the database"`
+	RPCHost            string `flag:"rpc-host" default:"localhost" info:"host for the RPC API server"`
+	RPCPort            int    `flag:"rpc-port" default:"3000" info:"port for the RPC API server"`
+	AccessNodeGRPCHost string `flag:"access-node-grpc-host" default:"localhost:3569" info:"host to the flow access node gRPC API"`
+	InitHeight         uint64 `flag:"init-height" default:"EmptyHeight" info:"init cadence block height from where the event ingestion will start. WARNING: you should only provide this if there are no existing values in the database, otherwise an error will be thrown"`
+	NetworkID          string `flag:"network-id" default:"testnet" info:"EVM network ID (testnet, mainnet)"`
+	Coinbase           string `flag:"coinbase" default:"" info:"coinbase address to use for fee collection"`
+	GasPrice           string `flag:"gas-price" default:"1" info:"static gas price used for EVM transactions"`
+	COAAddress         string `flag:"coa-address" default:"" info:"Flow address that holds COA account used for submitting transactions"`
+	COAKey             string `flag:"coa-key" default:"" info:"WARNING: do not use this flag in production! private key value for the COA address used for submitting transactions"`
+	CreateCOAResource  bool   `flag:"coa-resource-create" default:"false" info:"auto-create the COA resource in the Flow COA account provided if one doesn't exist"`
+}
+
+var flagGateway = gatewayFlag{}
+
+var gatewayCommand = &command.Command{
+	Cmd: &cobra.Command{
+		Use:     "gateway",
+		Short:   "Start the EVM gateway that exposes EVM RPC APIs",
+		Example: "flow evm gateway",
+	},
+	Flags: &flagGateway,
+	Run: func(
+		args []string,
+		globalFlags command.GlobalFlags,
+		logger output.Logger,
+		readerWriter flowkit.ReaderWriter,
+		flow flowkit.Services,
+	) (command.Result, error) {
+		cfg := &config.Config{
+			DatabaseDir:        flagGateway.DatabaseDir,
+			AccessNodeGRPCHost: flagGateway.AccessNodeGRPCHost,
+			RPCPort:            flagGateway.RPCPort,
+			RPCHost:            flagGateway.RPCHost,
+			InitHeight:         flagGateway.InitHeight,
+			CreateCOAResource:  flagGateway.CreateCOAResource,
+		}
+
+		if flagGateway.Coinbase == "" {
+			return nil, fmt.Errorf("coinbase EVM address required")
+		}
+		cfg.Coinbase = gethCommon.HexToAddress(flagGateway.Coinbase)
+		if g, ok := new(big.Int).SetString(flagGateway.GasPrice, 10); ok {
+			cfg.GasPrice = g
+		}
+
+		cfg.COAAddress = flowsdk.HexToAddress(flagGateway.COAAddress)
+		if cfg.COAAddress == flowsdk.EmptyAddress {
+			return nil, fmt.Errorf("invalid COA address value")
+		}
+
+		pkey, err := crypto.DecodePrivateKeyHex(crypto.ECDSA_P256, flagGateway.COAKey)
+		if err != nil {
+			return nil, fmt.Errorf("invalid COA key: %w", err)
+		}
+		cfg.COAKey = pkey
+
+		switch flagGateway.NetworkID {
```

Comment:
Can the Emulator be supported?

Reply:
@turbolent Yeah, certainly. All this time we used the Emulator as an Access Node backend. But there's no dedicated chain ID for it, so we might have to use `emulator.FlowEVMTestnetChainID`, both `testnet` & `mainnet`.